File: backend/app/schemas/admin_analytics.py
from datetime import date

from pydantic import BaseModel


class DailyCount(BaseModel):
    date: date
    count: int


class CategoryCount(BaseModel):
    category: str
    count: int


class TopDataset(BaseModel):
    title: str
    count: int


class AnalyticsResponse(BaseModel):
    requests_submitted: int
    approval_rate_pct: float
    total_downloads: int
    new_researchers: int
    requests_over_time: list[DailyCount]
    downloads_over_time: list[DailyCount]
    new_users_over_time: list[DailyCount]
    requests_by_status: dict[str, int]
    requests_by_category: list[CategoryCount]
    top_datasets_by_grants: list[TopDataset]

File: frontend/src/lib/types/admin-analytics.ts
// Mirrors backend/app/schemas/admin_analytics.py exactly — keep these two in sync.

export interface DailyCount {
  date: string;
  count: number;
}

export interface CategoryCount {
  category: string;
  count: number;
}

export interface TopDataset {
  title: string;
  count: number;
}

export interface AnalyticsResponse {
  requests_submitted: number;
  approval_rate_pct: number;
  total_downloads: number;
  new_researchers: number;
  requests_over_time: DailyCount[];
  downloads_over_time: DailyCount[];
  new_users_over_time: DailyCount[];
  requests_by_status: Record<string, number>;
  requests_by_category: CategoryCount[];
  top_datasets_by_grants: TopDataset[];
}
